processing/src/main/java/io/druid/query/MetricsEmittingExecutorService.java

```diff
@@ -51,17 +54,87 @@ protected ListeningExecutorService delegate()
   }
 
   @Override
-  public <T> ListenableFuture<T> submit(Callable<T> tCallable)
+  public <T> ListenableFuture<T> submit(final Callable<T> tCallable)
   {
+    final Callable<T> wrappedPrioritizedCallable = new PrioritizedCallable<T>()
+    {
+      @Override
+      public int getPriority()
+      {
+        if (tCallable instanceof PrioritizedCallable) {
+          return ((PrioritizedCallable) tCallable).getPriority();
+        } else {
+          return 0;
```

Comment:
currently the default priority is specific to the PrioritizedExecutorService, so we should not make any assumptions here.